ironfish/src/network/peers/peer.ts

```diff
@@ -176,6 +186,11 @@ export class Peer {
    */
   readonly onKnownPeersChanged: Event<[]> = new Event()
 
+  /**
+   * Fired when the peer should be banned
+   */
+  readonly onBanned: Event<[]> = new Event()
```

Comment:
It doesn't look like this is emitted-to anywhere, is it planned to be used later?

Reply:
Fixed, that prevented the entire thing from working. I'll write a test here in a second.